internal/k8s/vcf_controller.go

```diff
@@ -0,0 +1,181 @@
+		}
+	}
+
+	utils.AviLog.Infof("Got error while fetching avi-secret: %v", err)
+
+	if !c.ValidBSData() {
+		utils.AviLog.Infof("Running in a VCF Cluster, but valid Bootstrap CR not found, waiting .. ")
+		startSyncCh := make(chan struct{})
+		c.AddNCPBootstrapEventHandler(informers, stopCh, startSyncCh)
+	L:
+		for {
+			select {
+			case <-startSyncCh:
+				break L
+			case <-ctrlCh:
+				return
+			}
+		}
+	}
+	utils.AviLog.Infof("NCP Bootstrap CR found, continuing AKO initialization")
+	c.UpdateAviSecret()
+}
+
+func (c *AviController) UpdateAviSecret() error {
+	secretName, ns, username := lib.GetBootstrapCRData()
+	cs := c.informers.KubeClientIntf.ClientSet
+	cs.CoreV1().Secrets(ns).Get(context.TODO(), secretName, metav1.GetOptions{})
+
+	var ncpSecret *corev1.Secret
+	var err error
+	ncpSecret, err = cs.CoreV1().Secrets(ns).Get(context.TODO(), secretName, metav1.GetOptions{})
+	if err != nil {
+		utils.AviLog.Warnf("Failed to get secret, got err: %v", err)
+		return err
+	}
+
+	var aviSecret corev1.Secret
+	aviSecret.ObjectMeta.Name = lib.AviSecret
+	aviSecret.Data = make(map[string][]byte)
+	aviSecret.Data["authtoken"] = []byte(ncpSecret.Data["authToken"])
+	aviSecret.Data["username"] = []byte(username)
+
+	_, err = cs.CoreV1().Secrets(utils.GetAKONamespace()).Get(context.TODO(), lib.AviSecret, metav1.GetOptions{})
+	if k8serrors.IsNotFound(err) {
+		_, err = cs.CoreV1().Secrets(utils.GetAKONamespace()).Create(context.TODO(), &aviSecret, metav1.CreateOptions{})
+		if err != nil {
+			utils.AviLog.Warnf("Failed to create avi-secret, err: %v", err)
+			return err
+		}
+	}
+
+	_, err = cs.CoreV1().Secrets(utils.GetAKONamespace()).Update(context.TODO(), &aviSecret, metav1.UpdateOptions{})
+	if err != nil {
+		utils.AviLog.Warnf("Failed to update avi-secret, err: %v", err)
+		return err
+	}
+
+	return nil
+}
+
+func (c *AviController) ValidBSData() bool {
+	utils.AviLog.Infof("Validating BS data")
+	cs := c.informers.ClientSet
+	secretName, ns, username := lib.GetBootstrapCRData()
+	utils.AviLog.Infof("Got data from Bootstrap CR, secretName: %s, namespace: %s, username: %s", secretName, ns, username)
+	var ncpSecret *corev1.Secret
+	var err error
+	ncpSecret, err = cs.CoreV1().Secrets(ns).Get(context.TODO(), secretName, metav1.GetOptions{})
+	if err != nil {
+		utils.AviLog.Warnf("Failed to get secret, got err: %v", err)
+		return false
+	}
+	authToken := ncpSecret.Data["authToken"]
+	ctrlIP := os.Getenv(utils.ENV_CTRL_IPADDRESS)
+	var transport *http.Transport
+	_, err = clients.NewAviClient(
+		ctrlIP, username, session.SetAuthToken(string(authToken)),
+		session.SetNoControllerStatusCheck, session.SetTransport(transport),
+		session.SetInsecure,
+	)
+	if err != nil {
+		utils.AviLog.Infof("Failed to connect to AVI controller using existing AKO secret, err: %v", err)
+		return false
+	}
+	utils.AviLog.Infof("Successfully connected to AVI controller using existing AKO secret")
```

Comment:
I think message needs to be "Successfully connected to Avi controller using NCP secret"

Reply:
Yeah, the msg was wrong, updated